plaid/errors.py

```diff
@@ -11,14 +11,35 @@ class PlaidError(Exception):
                                         for programmatic use. May be None.
     :ivar   str     request_id:         A unique id returned for all server
                                         responses.
+    :ivar   list    causes:             A list of reasons explaining why the
+                                        error happened.
     '''
 
-    def __init__(self, message, type, code, display_message, request_id=""):
+    def __init__(
+        self,
+        message,
+        type,
+        code,
+        display_message,
+        request_id="",
+        causes=[],
```

Comment:
I think in general in python you want to avoid passing in {} or [] as a default argument. Because otherwise all callers of this function will default to the same instance, and if it's ever mutated bad things happen in the future.

Reply:
Yep, either make it a tuple, or use `None` and  do `self.causes = causes or []` below.

Reply:
⚡️